telethon/helpers.py

```diff
@@ -95,16 +95,19 @@ def strip_text(text, entities):
     return text
 
 
-def retry_range(retries, reconnect=False):
+def retry_range(retries, force_retry=True):
     """
     Generates an integer sequence starting from 1. If `retries` is
     not a zero or a positive integer value, the sequence will be
     infinite, otherwise it will end at `retries + 1`.
     """
 
-    if retries == 0 and not reconnect:
+    # We need atleast one iteration even if the retries are 0
+    # when force_retry is True. 
+    if retries == 0 and force_retry:
         yield 1
 
+    # If retries are non 0 then itetrate
     attempt = 0
     while attempt != retries:
         yield 1 + attempt
```

Comment:
You didn't fix this as I said, instead, you've made a redundant increment.

Reply:
I changed it to -

```
attempt += 1
yield attempt
```

and the other Special one I stated the reason